crates/binjs_io/src/entropy/rw.rs

```diff
@@ -160,10 +162,29 @@ pub type NameData = SmallVec<[u8; NAME_MAX_LEN]>;
 /// may depend on that of previous values.
 ///
 /// `TableRef` values are represented as follows:
-/// - a value `0` represents an `TableRef::Prelude` with index 1 + the latest `TableRef::Prelude` value we have encoded/decoded;
-/// - a value `v` in `[1, window len]` represents a repeat of a previous value: 1 it the latest value, 2 the second latest, etc.
-/// - a value `v` in `[window len + 1, window len + length of the shared dictionary]` represent `TableRef::Shared` with index `v - window len - 1`;
-/// - a value `v > window len + length of the shared dictionary` represents `TableRef::Prelude` with index `v - window len - length of shared dictionary - 1`.
+/// - a value `0` represents an `TableRef::Prelude` with index 1 + the
+///     latest `TableRef::Prelude` value we have encoded/decoded;
+/// - a value `v` in [1, window len], where `window len` is a length
+///     determining when constructing this `TableRefStreamState`,
+///     represents a repeat of a previous value:
+///       - 1 for the latest value;
+///       - 2 the second latest;
+///       - ...
+/// - a value `v` in [window len + 1, window len + prelude len] represents
+///     a `TableRef::Prelude` with index `v - window len - 1`;
+/// - a value `v` in [window len + prelude len + 1,
+///     window len + prelude len + import len], where `import len` is the number
+///     of distinct references to the shared dictionary encountered *so far*,
+///     represents a `TableRef::Shared` that has already been encountered at
+///     least once in this stream
+///       - window len + prelude len + 1 if it was the first `TableRef::Shared`
+///         encountered in this stream;
+///       - window len + prelude len + 2 if it was the second;
+///       - ...
+/// - a value `v > window len + prelude len + import len` represents the first
+///     instance of a `TableRef::Shared`. Its index in the shared dictionary
+///     is `v - (window len + prelude len + import len)`.
```

Comment:
I feel, describing in the opposite way (like `TableRef:Prelude(0)` maps to `window_len + 1`) is better

Reply:
There are now two descriptions/specifications, one for compression and one for decompression.

It's going to be lots of fun to update whenever I make further changes to the format :)